packages/react/src/reconciler/hostconfig.js

```diff
@@ -104,9 +104,15 @@ function insertBefore(parent, child, beforeChild)
     invariant(child !== beforeChild, 'pixi-react: PixiFiber cannot insert node before itself');
 
     const childExists = parent.children.indexOf(child) !== -1;
+
+    if (childExists)
```

Comment:
I've done it this way in the past too, but I'm thinking that since PixiJS removes the child from its former parent if one exists anyway, we can skip the more expensive index check entirely by instead checking for a `.parent` on the child and removing it from said parent if it exists.
```
if (child.parent) {
  child.parent.removeChild(child);
}
```

Reply:
FWIW, `addChildAt` already does the above, so maybe we can drop this check altogether?

Reply:
Unfortunately it's needed for the very next index check, since the beforeChild index may or may not change if the child is already on the parent.

Reply:
I considered this during review, `indexOf` perhaps isn't the most expensive thing, assuming a sensible number of children. But @lunarraid your suggestion also makes sense. Would the following be a safer check:

```
if(child.parent === parent) {
  parent.removeChild(child);
} 
```

Although I guess if the parent's don't match up there's a bigger problem going on 😄

Reply:
Have merged for now and we can circle back if we consider this worth changing